### PR TITLE
Fix overlay nexthop ref_count is incorrect issue when add and delete route at same time.

### PR DESCRIPTION
**Pull request by iris00522**
**What I did**
reference #1501 : remove overlay nexthop  after updating the reference counter for a bulk of routes

**Why I did it**
error log
~~~
Mar 16 06:35:48.090274 accton-leaf-demo-002-leaf1 NOTICE swss#orchagent: :- removeRoutePost: Remove overlay Nexthop 10.0.0.28@vniVlan1000@10000@04:f8:f8:6a:fa:91
Mar 16 06:35:48.090333 accton-leaf-demo-002-leaf1 NOTICE swss#orchagent: :- removeOverlayNextHops: Remove overlay Nexthop 10.0.0.28@vniVlan1000@10000@04:f8:f8:6a:fa:91
Mar 16 06:35:48.090333 accton-leaf-demo-002-leaf1 NOTICE swss#orchagent: :- removeNextHopTunnel: NH tunnel remove for vtep1, ip 10.0.0.28, mac 04:f8:f8:6a:fa:91, vni 10000
Mar 16 06:35:48.090376 accton-leaf-demo-002-leaf1 NOTICE swss#orchagent: :- removeNextHop: remove NH tunnel for ip 10.0.0.28, mac 04:f8:f8:6a:fa:91, vni 10000, ref_count 1
Mar 16 06:35:48.090394 accton-leaf-demo-002-leaf1 ERR swss#orchagent: :- meta_generic_validation_remove: object 0x4000000000a1f reference count is 1, can't remove
Mar 16 06:35:48.090394 accton-leaf-demo-002-leaf1 NOTICE swss#orchagent: :- removeNextHop: delete NH tunnel for ip '10.0.0.28', mac '04:f8:f8:6a:fa:91' vni 10000 failed
~~~
**How I verified it**
test on broadcom dut , retry connect and disconnect remote vtep more than 20 times.

### CODE REVIEW

File: orchagent/routeorch.cpp [hunk 1]
```diff
@@ -1896,10 +1903,10 @@ bool RouteOrch::removeRoutePost(const RouteBulkContext& ctx)
         if (it_route->second.getSize() > 1
```
Comment (shi-su) [inline orchagent/routeorch.cpp:1903]: Maybe we can replace all `it_route->second` with `ol_nextHops` if this `if ... else if ...` to make use of `ol_nextHops = it_route->second`

**Comment by lguohan** [comment]
can you add vstest to cover this case?

**Comment by iris00522** [comment]
Please help to review, thank you.

**Review by shi-su — APPROVED** [review]

**Comment by iris00522** [comment]
@shi-su , please help to merge codes. Thank you.

**Pull request merged**